File: src/main/java/com/elchologamer/userlogin/manager/PlayerManager.java
package com.elchologamer.userlogin.manager;

import com.elchologamer.userlogin.ULPlayer;
import org.bukkit.entity.Player;

import java.util.HashMap;
import java.util.UUID;

public class PlayerManager extends HashMap<UUID, ULPlayer> {

    public ULPlayer getOrCreate(Player player) {
        UUID uuid = player.getUniqueId();

        if (containsKey(uuid)) {
            return get(uuid);
        } else {
            ULPlayer newPlayer = new ULPlayer(player);
            put(uuid, newPlayer);

            return newPlayer;
        }
    }
}

File: src/main/kotlin/com/elchologamer/userlogin/manager/PlayerManager.kt
package com.elchologamer.userlogin.manager

import com.elchologamer.userlogin.ULPlayer
import org.bukkit.entity.Player
import java.util.*

class PlayerManager : HashMap<UUID, ULPlayer>() {
    fun getOrCreate(player: Player): ULPlayer {
        var ulPlayer: ULPlayer? = get(player.uniqueId)

        if (ulPlayer == null) {
            ulPlayer = ULPlayer(player)
            put(player.uniqueId, ulPlayer)
        }

        return ulPlayer
    }
}